azure-quantum/azure/quantum/optimization/online_problem.py

```diff
@@ -0,0 +1,33 @@
+import logging
+from typing import TYPE_CHECKING, Union, Dict 
+from azure.quantum.optimization import Problem
+
+logger = logging.getLogger(__name__)
+
+__all__ = ['OnlineProblem']
+
+if TYPE_CHECKING:
+    from azure.quantum.workspace import Workspace
+
+class OnlineProblem(Problem):
```

Comment:
As mentioned offline, we could consider the possibility of doing this the other way around - i.e. having Problem inherit OnlineProblem instead. This way we don't need to throw all those exceptions for unsupported methods.

Reply:
On second thought, maybe they should both inherit from a common base (?) and then OnlineProblem can still call Problem  - I am not sure, let me know your thoughts.

Reply:
+1 on having a ProblemBase that is an abstract class and have the Problem and OnlineProblem being descends of it.

Reply:
If the .download() functionality is giving you too much trouble, I'm happy to skip it and just let Problem.py take in a blob url or something if the user wants to explicitly download a blob into a Problem.py. In terms of just repeat submissions, I'm happy with OnlineProblem being a wrapper around a blob uri & name

Reply:
I suggest waiting for #53 before doing more changes because #53 has a lot of PEP8 fixes across many files.

Reply:
In order to resolve this, I have made OnlineProblem not inherit from problem and use Problem's download method to download it. Please let me know if this looks better now. 
We can visit the idea of having a base problem class and have all three types of problems (Problem, OnlineProbem and Streaing Problem inherit from it) but that might tie into larger changes more refactiring.